tests/test_rooms/test_assign_resource.py

```diff
@@ -0,0 +1,81 @@
+from tests.base import BaseTestCase, CommonTestCases
+from fixtures.token.token_fixture import ADMIN_TOKEN
+from fixtures.room.assign_resource_fixture import (
+    assign_resource_mutation,
+    assign_resource_mutation_response,
+    assign_resource_non_existent_room,
+    assign_non_existent_resource_id,
+    assign_excess_quantity,
+    assign_quantity_less_than_one,
+    query_string
+    )
+
+import sys
+import os
+sys.path.append(os.getcwd())
+
+
+class TestAssignResource(BaseTestCase):
+
+    def test_assign_resource(self):
+        """
+            Test for assigning a resource to a room
+        """
+        CommonTestCases.admin_token_assert_equal(
+           self,
+           assign_resource_mutation,
+           assign_resource_mutation_response,
+        )
+
+    def test_assign_non_existent_resource(self):
+        """
+            Test assigning a resource that does not exist
+        """
+        CommonTestCases.admin_token_assert_in(
+            self,
+            assign_non_existent_resource_id,
+            'invalid resource id'
+        )
+
+    def test_assign_non_existent_room(self):
+        """
+            Test assigning a resource to a room that does not exist
+        """
+        CommonTestCases.admin_token_assert_in(
+            self,
+            assign_resource_non_existent_room,
+            'Room not found'
+        )
+
+    def test_assign_resource_excess_quantity(self):
+        """
+            Test assign a resource with a quantity higher than\
+            exists in database
+        """
```

Comment:
@d-kahara Just a little nitpicking:
can you change `higher than exists in database` to `higher than quantity in database` or `higher than available quantity`

Reply:
Nice suggestion @rajeman . I think `higher than available quantity` is more user oriented.